### DISCUSSION TITLE
Добро пожаловать в  Sberbank2Excel дискуссию!

### DISCUSSION THREAD

**Discussion by Ev2geny** [Ev2geny/Sberbank2Excel#10] [Announcements]
Добро пожаловать в  Sberbank2Excel дискуссию!

**Comment by Kribli-krabli** [comment]
Работает. Но иногда выдает ошибку:

Конвертируем файл C:/Users/****/Downloads/Выписка по дебетовой карте (на русском) (1).pdf
Формат файла определён как 2107_Stavropol
Произошла ошибка при конвертации файла "file" <class 'core.exceptions.BalanceVerificationError'>
Traceback (most recent call last):
  File "core\sberbankPDF2ExcelGUI.py", line 56, in btn_convertFiles_clicked
  File "core\sberbankPDF2Excel.py", line 23, in sberbankPDF2Excel
  File "core\sberbankPDFtext2Excel.py", line 57, in sberbankPDFtext2Excel
  File "core\utils.py", line 425, in check_transactions_balance
core.exceptions.BalanceVerificationError:
            Ошибка проверки балланса по трансакциям:
                СУММА НАЧИСЛЕНИЙ - СУММА СПИСАНИЙ = 54392.57
                Вычисленный баланс по всем трансакциям = 54452.57

**Reply by Kribli-krabli** [reply]
![image](https://user-images.githubusercontent.com/45893971/126640639-0b61531b-82e7-4fb9-8818-71a98a6d0017.png)

**Comment by Ev2geny** [comment]
У вас странный формат выписки - отсутствует код авторизации. Обычно либо 5
цифр, либо прочерк, а в вашей выписке он вообще отсутствует.

Есть, какие-то наблюдения когда появляется такая выписка?

**Reply by Kribli-krabli** [reply]
Я его просто закрасил :)

**Comment by Ev2geny** [comment]
Тяжело будет исправить не имея конкретного  pdf файла для тестирования.
Мне, наверное,  придётся создать версию, где не удаляется промежуточная
текстовая версия ( pdf сначала конвертируется в текстовый формат) чтобы
люди могли бы удалять из него конфиденциальную информацию и потом
отправлять для исправления утилиты.

**Reply by Ev2geny** [reply]
@Kribli-krabli , пожалуйста проверьте, что вы используете версию https://github.com/Ev2geny/Sberbank2Excel/releases/tag/v3.1.1

**Reply by Ev2geny** [reply]
Выпущена версия, которая имеет функцию создания промежуточного текстового файла.
https://github.com/Ev2geny/Sberbank2Excel/releases/tag/v.3.3.0
Если нет возможности переслать разработчику изначальную выписку из соображений конфиденциальности, можно переслать анонимизированный промежуточный текстовый файл. 
Инструкции здесь
https://github.com/Ev2geny/Sberbank2Excel

**Comment by Latoid** [comment]
Добрый день.
Спасибо за отличную программу. Просто спасение для людей, кто любит контролировать свои расходы.
И просьба, если возможно:
добавить в GUI новую галочку: "Обратить порядок платёжных событий". Чтоб при выборе её результат в Excel получался бы не "вверху самые новые транзакции, внизу - самые старые", а наоборот "вверху самые старые транзакции, внизу - самые новые".

Я знаю, что это легко можно сделать в Excel, но всё же это дополнительные телодвижения: добавить новую колонку "№ строки", проставить в ней числа по возрастанию, добавить сортировку на всю таблицу, и отсортировать добавленную колонку в обратном порядке. Но всё же - это дополнительные рутинные операции для пользователя, что занимают у него, скажем, минуту времени по каждой выписке (а их много за один раз бывает). Тогда как программа всё тоже самое сделает за долю секунды.

**Reply by alexandrdera** [reply]
Добрый день! Что то вы сильно усложнили, отсортируйте строки по дате операции как вам нужно и всё.

**Reply by Ev2geny** [reply]
Здравствуйте,

спасибо за обратную связь. Добавлю функционал, это несложно. Я создал новую "issue" для этого
https://github.com/Ev2geny/Sberbank2Excel/issues/29#issue-1657110884

Если есть какие-то дополнительные комментарии по этому функционалу - лучше писать туда.

Пару вопросов (просто ради нтререса): 

- Используете ли вы какие-то другие программы для контроля расходов, или просто делаете всё в Excel?

- Почему так получается, что у вас бывает сразу много выписок?  у вас сразу много счетов / карточек?

**Reply by Latoid** [reply]
Спасибо. Это будет здОрово.
Ответы на вопросы:
1. Использую только Excel. Меня устраивает, альтернатив особо не искал (хотя, может, и стоило бы).
2. У меня самого три карточки (Сбер в своё время навыдавал) плюс слежу ежемесячно за расходами на двух карточках своих пожилых родителей. А то, бывает, подпишутся на какую-то хрень в Интернете, а потом мне вопросы - куда деньги с карты списываются.

**Comment by Latoid** [comment]
Вижу, вы выпустили новый коммит.
Ещё не тестил, но спасибо большое!

**Reply by Ev2geny** [reply]
Здравствуйте, удалось ли проверить новый функционал?
https://github.com/Ev2geny/Sberbank2Excel/releases/tag/v.4.5.0

**Reply by Latoid** [reply]
Здравствуйте.
Да, всё отлично, спасибо вам большое

**Comment by Ev2geny** [comment]
Подождите, я еще не загрузил релиз

**Reply by Ev2geny** [reply]
Всё, можно скачивать,[ v.4.5.0](https://github.com/Ev2geny/Sberbank2Excel/releases/tag/v.4.5.0)

**Comment by Trigun27** [comment]
Привет! Спасибо за прогу!
Надеюсь майнеров нет? 

В любом случае, вы меня спасли от мучительного ручного набора транзакций в beancount.

**Reply by Ev2geny** [reply]
Здравствуйте.

Спасибо за обратную связь.

Майнеров нет

Я, кстати, тоже перевожу все это в beancount.
Какой механизм вы используете для создания beancount формата из Excel выписки?

**Reply by Trigun27** [reply]
Перевожу в csv. 
а для csv в beancount мне пока хватает дженерик встроенного конвертера. 
Только надо задать в параметрах для Парсинга даты, что FirstDay=True, чтобы даты правильно парсились. А то вместо 04.01.20204 получаешь  1 апреля 2024 года.

**Comment by Nikkotor** [comment]
Привет! 

Сегодня попробовал сконвертировать выписк, но получил отлуп. возможно сбер что-то поменял.

Traceback (most recent call last):
  File "sberbankPDF2ExcelGUI.py", line 63, in btn_convertFiles_clicked
  File "sberbankPDF2Excel.py", line 66, in sberbankPDF2Excel
  File "sberbankPDFtext2Excel.py", line 82, in sberbankPDFtext2Excel
  File "extractors_generic.py", line 26, in determine_extractor_auto
exceptions.InputFileStructureError: Неизвecтный формат выписки, ни один из экстракторов не подходят

Пропускаем конвертацию этого файла
!!!!!!! 1 файл(а) из 1 не были сконвертированы

**Comment by Ev2geny** [comment]
Добрый день,

- убедитесь пожалуйста, что вы используете [последнюю версию программы](https://github.com/Ev2geny/Sberbank2Excel/releases/latest)

- Если первый шак не помог, сообщите какую выписку вы пытаетесь сконветировать (на что она похоже из поддерживаемых). См. [Приложение А. Список поддерживаемых форматов](https://github.com/Ev2geny/Sberbank2Excel#%D0%BF%D1%80%D0%B8%D0%BB%D0%BE%D0%B6%D0%B5%D0%BD%D0%B8%D0%B5-%D0%B0-%D1%81%D0%BF%D0%B8%D1%81%D0%BE%D0%BA-%D0%BF%D0%BE%D0%B4%D0%B4%D0%B5%D1%80%D0%B6%D0%B8%D0%B2%D0%B0%D0%B5%D0%BC%D1%8B%D1%85-%D1%84%D0%BE%D1%80%D0%BC%D0%B0%D1%82%D0%BE%D0%B2). Я попытаюсь воспроизвести эту проблему на своих выписках.

**Reply by Nikkotor** [reply]
Разобрался,  все работает. Оказывается, заказывал так называемую "индивидуальную выписку", она имеет другой формат. С обычной выпиской все получилось.  Кстати, пожелание с точки зрения юзабилити - по результатам работы программы, выводится строчка с путем к итоговому файлу. При этом неочевидно, что программа завершила работу. Я бы посоветовал добавить кнопки типа - "новый файл" и "принять и закрыть", которые становились бы активными после обработки текущего задания.

**Reply by disinvis** [reply]
-

**Comment by disinvis** [comment]
<img width="978" height="632" alt="image" src="https://github.com/user-attachments/assets/5e9a6590-17d5-41d1-a2ea-98b905c31cf6" />
<img width="842" height="592" alt="image" src="https://github.com/user-attachments/assets/a5a1abf5-3509-4b99-90d3-ca3824d5f45d" />
Это автоплатёж за квартиру. То ли больше строк в отчёте занимает, то ли ещё отчего-то, но выдаёт ошибку.

**Comment by Ev2geny** [comment]
Спасибо за сообщение об ошибке. 

Я создал новую проблему  на основе вашей информации https://github.com/Ev2geny/Sberbank2Excel/issues/67

Дальнейшее обсуждение по этой теме давайте вести там.

Для успешной работы над проблемой очень бы хотелось получить если не оригинал выписки, то хотя бы анонимизированный промежуточный текстовый файл. Как его создать написано в [приложении Б](https://github.com/Ev2geny/Sberbank2Excel?tab=readme-ov-file#%D0%BF%D1%80%D0%B8%D0%BB%D0%BE%D0%B6%D0%B5%D0%BD%D0%B8%D0%B5-%D0%B1-%D0%BA%D0%B0%D0%BA-%D0%B1%D0%B5%D0%B7%D0%BE%D0%BF%D0%B0%D1%81%D0%BD%D0%BE-%D0%BF%D0%B5%D1%80%D0%B5%D1%81%D1%8B%D0%BB%D0%B0%D1%82%D1%8C-%D0%BF%D1%80%D0%BE%D0%B1%D0%BB%D0%B5%D0%BC%D0%BD%D1%8B%D0%B9-%D1%84%D0%B0%D0%B9%D0%BB)

Для простоты просто создайте выписку за 1 день (7 июля)

**Comment by Timur-san** [comment]
Доброго дня,
обычно работал, но теперь вот...

> Версия 5.0.1
> format='auto'
> ******************************
> Конвертируем файл C:/Users…СберБанк/20251125 Выписка по счёту кредитной карты.pdf
> Формат файла определён как SBER_CREDIT_2110
> Произошла ошибка при конвертации файла C:/Users/…СберБанк/20251125 Выписка по счёту кредитной карты.pdf
> <class 'RuntimeError'>
> Traceback (most recent call last):
>   File "Sberbank2Excel\extractor.py", line 99, in get_entries
>   File "Sberbank2Excel\extractor_SBER_CREDIT_2110.py", line 182, in decompose_entry_to_dict
> Sberbank2Excel.exceptions.InputFileStructureError: entry is expected to have from 2 to 3 lines
> 20.10.2025      14:23   Перевод на карту        +15 000,00      299 901,42
> 20.10.2025      401186  Перевод от Г.  Тимур Надариевич. Операция по карте
> ****7942
> *
> Дата формирования документа     25.11.2025
> 40601D00C08FCE2CD999F93A68651986
> с 02.07.2025 по 02.10.2026
> 25.11.2025
> 
> 
> The above exception was the direct cause of the following exception:
> 
> Traceback (most recent call last):
>   File "Sberbank2Excel\sberbankPDF2ExcelGUI.py", line 73, in btn_convertFiles_clicked
>   File "Sberbank2Excel\sberbankPDF2Excel.py", line 67, in sberbankPDF2Excel
>   File "Sberbank2Excel\sberbankPDFtext2Excel.py", line 103, in sberbankPDFtext2Excel
>   File "Sberbank2Excel\extractor.py", line 108, in get_entries
> RuntimeError: Ошибка при обработке трансакции
> --------------------
> 20.10.2025      14:23   Перевод на карту        +15 000,00      299 901,42
> 20.10.2025      401186  Перевод от Г.  Тимур Надариевич. Операция по карте
> ****7942
> *
> Дата формирования документа     25.11.2025
> 40601D00C08FCE2CD999F93A68651986
> с 02.07.2025 по 02.10.2026
> 25.11.2025
> 
> --------------------
> 
> Пропускаем конвертацию этого файла
> !!!!!!! 1 файл(а) из 1 не были сконвертированы

Спотыкается всегда на первой по хронологии транзакции. Пробовал выгрузки с разными датами – везде та же ошибка.

**Reply by Ev2geny** [reply]
Здравствуйте, к сожалению я сам не использую кредитные карты, а для успешной работы над проблемой хотелось бы получить если не оригинал выписки, то хотя бы анонимизированный промежуточный текстовый файл. Как его создать написано в [приложении Б](https://github.com/Ev2geny/Sberbank2Excel?tab=readme-ov-file#%D0%BF%D1%80%D0%B8%D0%BB%D0%BE%D0%B6%D0%B5%D0%BD%D0%B8%D0%B5-%D0%B1-%D0%BA%D0%B0%D0%BA-%D0%B1%D0%B5%D0%B7%D0%BE%D0%BF%D0%B0%D1%81%D0%BD%D0%BE-%D0%BF%D0%B5%D1%80%D0%B5%D1%81%D1%8B%D0%BB%D0%B0%D1%82%D1%8C-%D0%BF%D1%80%D0%BE%D0%B1%D0%BB%D0%B5%D0%BC%D0%BD%D1%8B%D0%B9-%D1%84%D0%B0%D0%B9%D0%BB)

**Reply by Timur-san** [reply]
Отправил на почту

**Reply by Ev2geny** [reply]
Спасибо. Получил. Сделаем!

**Reply by Ev2geny** [reply]
Исправлено в релизе [5.1.0](https://github.com/Ev2geny/Sberbank2Excel/releases/tag/v5.1.0)

**Comment by Timur-san** [comment]
Приветствую!
Опять перестали конвертиться выписки, кредитная и дебетовая (
5.3.1
Выслал файлы почтой.

**Reply by Ev2geny** [reply]
Ок, посмотрим. Создал проблему
https://github.com/Ev2geny/Sberbank2Excel/issues/87
